protocol/xmrmaker/swap_state.go

```diff
@@ -136,31 +142,68 @@ func newSwapState(
 		walletScanHeight -= monero.MinSpendConfirmations
 	}
 
-	ctx, cancel := context.WithCancel(b.Ctx())
-	s := &swapState{
-		ctx:                 ctx,
-		cancel:              cancel,
-		Backend:             b,
-		sender:              sender,
-		offer:               offer,
-		offerExtra:          offerExtra,
-		offerManager:        om,
-		walletScanHeight:    walletScanHeight,
-		nextExpectedMessage: &net.SendKeysMessage{},
-		readyCh:             make(chan struct{}),
-		info:                info,
-		done:                make(chan struct{}),
+	ethHeader, err := b.EthClient().HeaderByNumber(b.Ctx(), nil)
+	if err != nil {
+		return nil, err
 	}
 
-	return s, nil
-}
+	// set up ethereum event watchers
+	logReadyCh := make(chan []ethtypes.Log)
+	logRefundedCh := make(chan []ethtypes.Log)
 
-func (s *swapState) lockState() {
-	s.stateMu.Lock()
-}
+	ctx, cancel := context.WithCancel(b.Ctx())
 
-func (s *swapState) unlockState() {
-	s.stateMu.Unlock()
+	readyWatcher := watcher.NewEventFilterer(
+		ctx,
+		b.EthClient(),
+		b.ContractAddr(),
+		ethHeader.Number,
+		readyTopic,
+		logReadyCh,
+	)
+
+	refundedWatcher := watcher.NewEventFilterer(
+		ctx,
+		b.EthClient(),
+		b.ContractAddr(),
+		ethHeader.Number,
+		refundedTopic,
+		logRefundedCh,
+	)
```

Comment:
How do these [removed log events](https://github.com/ethereum/go-ethereum/blob/v1.10.26/core/types/log.go#L53-L55) and the [`Removed` field](https://github.com/ethereum/go-ethereum/blob/v1.10.26/core/events.go#L30-L31) in the actual logs work? Do we need to be checking that field somewhere?

Reply:
hm, yeah seems that should be checked in case of re-org, will update